kernel_hardening_checker/engine.py

```diff
@@ -81,6 +81,17 @@ def set_state(self, data: StrOrNone) -> None:
         self.state = data
 
     def check(self) -> None:
+        # handle checks, provided with list()
+        if self.expected.startswith('*') and self.expected.endswith('*'):
```

Comment:
Please rework this block of code carefully.
You can avoid unneeded indent.
You also should not proceed to the next code after handling this type of checks. This is a bug.